pkg/csi/service/common/commonco/types/types.go

```diff
@@ -33,15 +33,64 @@ type NodeInfo struct {
 	NodeID string
 }
 
+// VanillaTopologyFetchDSParams represents the params required to call
+// GetSharedDatastoresInTopology in vanilla cluster.
+type VanillaTopologyFetchDSParams struct {
+	// TopologyRequirement represents the topology conditions
+	// which need to be satisfied during volume provisioning.
+	TopologyRequirement *csi.TopologyRequirement
+}
+
+// WCPTopologyFetchDSParams represents the params required to call
+// GetSharedDatastoresInTopology in workload cluster.
+type WCPTopologyFetchDSParams struct {
+	// TopologyRequirement represents the topology conditions
+	// which need to be satisfied during volume provisioning.
+	TopologyRequirement *csi.TopologyRequirement
+	// Vc is the vcenter instance using which the potential
+	// datastores will be calculated.
+	Vc *cnsvsphere.VirtualCenter
```

Comment:
during the creation of a new vCenter instance due to session time out or configuration reload, can we make sure this is also getting updated?

Reply:
Yes a VC instance is created on each CreateVolume call, so this will be reloaded when the session times out.